pkg/providers/cloudstack/cloudstack.go

```diff
@@ -279,7 +279,22 @@ func (p *cloudstackProvider) UpdateKubeConfig(_ *[]byte, _ string) error {
 }
 
 func (p *cloudstackProvider) BootstrapClusterOpts() ([]bootstrapper.BootstrapClusterOption, error) {
-	return common.BootstrapClusterOpts(p.datacenterConfig.Spec.ManagementApiEndpoint, p.clusterConfig)
+	env := map[string]string{}
```

Comment:
I would keep using the original `common.BootstrapClusterOpts` and then add cloudstack custom NO_PROXY here to avoid duplicate code.

I'd also add clusterSpec as input arg, so that you can use clusterSpec.CloudStackDatacenter directly and remove https://github.com/aws/eks-anywhere/blob/18662f1c337379f49753a0e377b6dcdd83fa73d9/pkg/providers/cloudstack/cloudstack.go#L259 completely from the struct.

It is ok to change the interface method. For any other provider, just do
`func (p *vsphereProvider) BootstrapClusterOpts(_ *cluster.Spec) ([]bootstrapper.BootstrapClusterOption, error)`